pkg/centralconfig/central_config.go

```diff
@@ -0,0 +1,39 @@
+// Copyright 2024 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Package centralconfig implements an interface to deal with the central configuration.
+package centralconfig
+
+import (
+	"path/filepath"
+
+	"github.com/vmware-tanzu/tanzu-cli/pkg/common"
+	"github.com/vmware-tanzu/tanzu-plugin-runtime/config/types"
+)
+
+// CentralConfigFileName is the name of the central config file
+const CentralConfigFileName = "central_config.yaml"
+
+// CentralConfigEntryKey represents the key of a central configuration entry.
+type CentralConfigEntryKey struct {
+	Key string
+}
+
+// CentralConfigEntryValue represents the value of a central configuration entry.
+type CentralConfigEntryValue struct {
+	Value interface{}
+}
```

Comment:
nit:
I am trying to understand the reasoning behind creating a wrapper here instead of just accepting `key string` as an input and returning `value interface{}` directly. 

Are we thinking about adding more metadata with the `CentralConfigEntryKey` and `CentralConfigEntryValue`?

Reply:
I did this because I was thinking that if we want to provide this API to plugins, then having a type, as you say, allows to add more fields in a backwards-compatible way.

That being said, I do find it heavy and it may stay in the CLI forever, in which case we don't care about backwards-compatibility.

I'm ok to simplify it and not use a wrapper.  Let me know if you agree.